TemplePlus/feat.cpp

```diff
@@ -259,10 +260,26 @@ void LegacyFeatSystem::_GetNewFeatsFromFile()
 					
 					if (prereqArgCount < prereqCount)
 					{
-						featSpec.prereqs[prereqArgCount++].featPrereqCodeArg = atol(ch);
+						if (featSpec.prereqs.capacity() <= prereqArgCount)
+						{
+							featPrereq.featPrereqCodeArg = atol(ch);
+							featSpec.prereqs.push_back(featPrereq);
+							prereqArgCount++;
+						}
+						else
+							featSpec.prereqs[prereqArgCount++].featPrereqCodeArg = atol(ch);
 					}
 					else
-						featSpec.prereqs[prereqCount++].featPrereqCode = atol(ch);
+					{
+						if (featSpec.prereqs.capacity() <= prereqCount)
```

Comment:
You're right, size was the correct value I wanted. I just made a commit to this pull, removed the featSpec allocation at the start of the function, and used size/resize to cut down code to just 4 lines.